WordPress/gutenberg
# Interactivity API: Add `block supports` for `clientNavigation` and `interactive` properties on `block.json` schema.

## What?

Add `clientNavigation` and `interactive` properties to block.json schema.

## Why?

When the Interactivity API is public, developers will need to indicate if their blocks are client side navigation compatible and interactive in their respective block.json files.

Client navigation will be opt out. Developers will need to set it as true if they want their block to work in client side navigation. (This means there is no interactivity or the block uses the Interactivity API to add interactions).

## Testing Instructions




- Point the schema to block.json schema instead of the default one.
- Create an interactive block.
- Check that schemas appear on the editor.
- Set your block to clientNavigation as true, put it inside a query block and disable "force reload". Check that client side navigation happens (browser does not reload).
- Set your block to clientNavigation as false, put it inside a query block and disable "force reload". Check that client side navigation does not happen (browser reloads).
